doc/Overview/Mvux/Advanced/Pagination.md

```diff
@@ -0,0 +1,398 @@
+
+An alternative way to paginate data is by using a cursor that points to a specific record and take then number of desired records in a page onwards.  
+This is referred to as 'keyset pagination' or 'seek-based pagination'.
+
+To utilize this pagination style, MVUX provides another `ListFeed` factory overload, the `AsyncPaginatedByCursor`.
+
+The signature of this method is:
+
+```csharp
+public static IListFeed<T> AsyncPaginatedByCursor<TCursor>(TCursor firstPage, GetPage<TCursor, T> getPage)
+```
+
+where the `GetPage` type is a delegate with the following pattern:
+
+```
+public delegate ValueTask<PageResult<TCursor, TItem>> GetPage<TCursor, TItem>(TCursor cursor, uint? desiredPageSize, CancellationToken ct);
+```
+
+The callback delegate takes two parameters and returns two.
+
+The two input parameters are:
+
+- A cursor to the first page, in other words the item this page starts with. The service then starts by ordering the data and searching for the first item, then uses the second parameter:
+- The desired page size, the number of entries following it, along with another entity that uses as the cursor of the next page, which is not displayed to the user.
+- A `CancallationToken` which can be used to abort the operation.
+
+The callback returns a `PageResult<TCursor, TItem>`, and it contains:
+
+- An `ImmutableList<T>` containing the items to be displayed in the resulted page.
+- A `TCursor` that can be of a different type that points to the beginning of the upcoming page.
+
+To demonstrate this we'll stick to the *PaginatedPeopleApp* we've started before.
+
+### Service
+
+To keep the service agnostic, we'll add the following method to the Service:
+
+*IPeopleService*:
+
+```csharp
+ValueTask<(IImmutableList<Person> CurrentPage, Person NextCursor)> GetPeopleAsync(Person? cursor, uint pageSize, CancellationToken ct);
+```
+
+This method has 3 parameters:
+
+- The first is a `Person` which uses as a cursor to the first page, i.e. the first `Person` of the requested page.
+- The second is a `uint` representing the number of entities to follow the first one. The reason of using a `uint` is that negative values are not expected here.
+- The third one is a `CancellationToken` which can be used to signal the service to abort loading data.
+
+The method returns a tuple with two components (same as `PageResult`, only keeping the service agnostic of MVUX):
+
+- An `IImmutableList<Person>` which includes the entities of this page
+- A `Person` which uses as a cursor as the beginning of the next page.
+
+> [!TIP]  
+> The cursor does not necessarily have to be a `Person` or the data type the collection contains, it can also be an ID or another key of an entity for the service to lookup and return it along with its upcoming entries.
+
+In the example above we want to override the `ToString` method of `Person` so that we can later order the entities by their string representation:
```

Comment:
```suggestion
This method has 3 parameters:

- The first is a `Person` which uses as a cursor to the first page, i.e. the first `Person` of the requested page.
- The second is a `uint` representing the number of entities to follow the first one. The reason for using a `uint` is that negative values are not expected here.
- The third one is a `CancellationToken` which can be used to signal the service to abort loading data.

The method returns a tuple with two components (same as `PageResult`, only keeping the service agnostic of MVUX):

- An `IImmutableList<Person>` which includes the entities of this page
- A `Person` which uses as a cursor at the beginning of the next page.

> [!TIP]  
> The cursor does not necessarily have to be a `Person` or the data type the collection contains, it can also be an ID or another key of an entity for the service to look up and return it along with its upcoming entries.

In the example above we want to override the `ToString` method of `Person` so that we can later order the entities by their string representation:
```